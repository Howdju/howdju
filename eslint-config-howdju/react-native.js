/**
 * ESLint config for React Native
 *
 * Use this instead of
 * [`eslint-config-react-native-community`](https://github.com/facebook/react-native/blob/main/packages/eslint-config-react-native-community/index.js).
 * While that config provides a useful starting point, it extends eslint-plugin-prettier, which is
 * [not recommended](https://prettier.io/docs/en/integrating-with-linters.html#notes). Instead,
 * we can use `prettier --check`.
 */
module.exports = {
  plugins: ["react-native", "@react-native-community"],
  extends: ["eslint-config-howdju/client-common", "prettier"],
};
