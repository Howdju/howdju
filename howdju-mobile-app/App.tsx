/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * Generated with the TypeScript template
 * https://github.com/react-native-community/react-native-template-typescript
 *
 * @format
 */

import React, {useCallback, useEffect, useState} from 'react';
import {
  Image,
  SafeAreaView,
  ScrollView,
  StatusBar,
  StyleSheet,
  Text,
  useColorScheme,
  View,
} from 'react-native';

import {
  Colors,
  Header,
} from 'react-native/Libraries/NewAppScreen';
import ShareMenu from 'react-native-share-menu';

type SharedItem = {
  mimeType: string;
  data: string;
  extraData: any;
};

const Section: React.FC<{
  title: string;
}> = ({children, title}) => {
  const isDarkMode = useColorScheme() === 'dark';
  return (
    <View style={styles.sectionContainer}>
      <Text
        style={[
          styles.sectionTitle,
          {
            color: isDarkMode ? Colors.white : Colors.black,
          },
        ]}>
        {title}
      </Text>
      <Text
        style={[
          styles.sectionDescription,
          {
            color: isDarkMode ? Colors.light : Colors.dark,
          },
        ]}>
        {children}
      </Text>
    </View>
  );
};

const App = () => {
  const isDarkMode = useColorScheme() === 'dark';

  const backgroundStyle = {
    backgroundColor: isDarkMode ? Colors.darker : Colors.lighter,
  };

  const [sharedData, setSharedData] = useState('');
  const [sharedMimeType, setSharedMimeType] = useState('');
  const [sharedExtraData, setSharedExtraData] = useState(null);

  const handleShare = useCallback((item: SharedItem | null) => {
    if (!item) {
      return;
    }

    const {mimeType, data, extraData} = item;

    setSharedData(data);
    setSharedExtraData(extraData);
    setSharedMimeType(mimeType);
  }, []);

  useEffect(() => {
    ShareMenu.getInitialShare(handleShare);
  }, [handleShare]);

  useEffect(() => {
    const listener = ShareMenu.addNewShareListener(handleShare);
    return () => {
      listener.remove();
    };
  }, [handleShare]);

  return (
    <SafeAreaView style={backgroundStyle}>
      <StatusBar barStyle={isDarkMode ? 'light-content' : 'dark-content'} />
      <ScrollView
        contentInsetAdjustmentBehavior="automatic"
        style={backgroundStyle}>
        <Header />
        <View
          style={{
            backgroundColor: isDarkMode ? Colors.black : Colors.white,
          }}>
          <Section title="Shared type">{sharedMimeType}</Section>
          <Section title="Shared data">{sharedData}</Section>
          <Section title="Shared image">
            {sharedMimeType.startsWith('image/') && (
              <Image
                style={styles.image}
                source={{uri: sharedData}}
                resizeMode="contain"
              />
            )}
          </Section>
          <Section title="Shared file">
            {sharedMimeType !== 'text/plain' &&
            !sharedMimeType.startsWith('image/')
              ? sharedData
              : ''}
          </Section>
          <Section title="Extra data">
            {sharedExtraData ? JSON.stringify(sharedExtraData) : ''}
          </Section>
        </View>
      </ScrollView>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  sectionContainer: {
    marginTop: 32,
    paddingHorizontal: 24,
  },
  sectionTitle: {
    fontSize: 24,
    fontWeight: '600',
  },
  sectionDescription: {
    marginTop: 8,
    fontSize: 18,
    fontWeight: '400',
  },
  highlight: {
    fontWeight: '700',
  },
  image: {
    width: '100%',
    height: 200,
  },
});

export default App;
