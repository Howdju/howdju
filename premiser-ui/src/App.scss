@import './reset';
@import './util';
@import './colors';
@import './react-md-custom';
@import './grid';

@include apply-react-md-grid;

html, html p {
  font-size: 15px;
}
html, a {
  color: $primary-text-color;
}
p a {
  color: $link-color;
}
// Display link buttons with the correct color
.md-btn.md-background--primary a {
  color: $md-primary-background-text-color;
}
.md-btn.md-background--secondary a {
  color: $md-secondary-background-text-color;
}

.error-message,
.error-message li {
  color: $md-text-field-error-color;
}

b {
  font-weight: 400;
}

h1, h2, h3, h4, h5, h6 {
  font-family: "Oswald", sans-serif;
  font-weight: 200;
}
h1 {
  font-size: 40px;
}

#webpack-hot-middleware-clientOverlay * {
  color: $light-text-color;
}

#app-nav-drawer {
  width: min(400px, $md-drawer-desktop-width)

  // TODO make this more like the left-nav rules
  // The left nav drawer's rules are:
  // On mobile devices, the width of the drawer will be calc(100vw - 56px) and a max width of 320px.
  // On tablets and desktop screens, the width will be equal to the $md-drawer-tablet-width and $md-drawer-desktop-width but no more than 400px.

}

// This should be in react-md's _tooltips.scss, but it isn't loaded for some reason
.inline-rel-container {
  display: inline-block;
  margin: 1em;
  position: relative;
}

// Tabs that appear to be contiguous with the toolbar.  But to avoid handling the navigate ourselves, they are actually not in the toolbar
.toolbarTabs {
  * {
    color: $light-text-color;
  }
}

.app-nav-drawer-header {
  @include center-contents;
}

.cell--centered-contents {
  @include center-contents;
}

.context-menu--floating {
  // Must be above the .justification-tree .verified icons
  z-index: 2;
  button {
    background-color: white;
  }
  &, &.md-menu-container {
    position: absolute;
  }
}

/* When the cards are vertically aligned, the spacing just distracts */
.md-grid--card-list {
  .md-cell {
    margin: 0;
  }
}

$small-chat-iframe-z-index: 999999999;
@media (max-width: $md-grid-desktop-breakpoint - 1) {
  // Two different ways to try and make sure the context menu is not obscured by the small chat launcher
  //#Smallchat iframe {
  //  right: auto !important;
  //  left: 0;
  //}
  //.context-menu {
  //  z-index: $small-chat-iframe-z-index + 1;
  //}

  // For cells that switch to single-column, we want no spacing
  .md-grid--card-list--tablet {
    .md-cell {
      margin: 0;
    }
  }
}

// Hide the small chat launcher when a no-scroll dialog is present on mobile
.md-no-scroll {
  #Smallchat {
    display: none;
  }
}

.md-inline-block.text-expand-toggle {
  vertical-align: text-top;
}

.clickable {
  cursor: pointer;
  &:hover {
    text-decoration: underline;
  }
}

.md-tab-label {
  font-family: "Oswald", sans-serif;
  font-weight: 200;
  font-size: 20px;
}

.selection-label {
  display: flex;
  align-items: center;

  .selection-label--text {
    margin-left: 5px;
  }
}

.entity-status-text {
  display: inline-block;
  color: $asbestos;
}