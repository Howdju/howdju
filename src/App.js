import React, { Component, PropTypes } from 'react'
import { Route, IndexRoute } from 'react-router'
import { BrowserRouter as Router } from 'react-router-dom'
import Helmet from 'react-helmet'
import Header from './Header'
import Home from './Home'
import StatementJustifications from './StatementJustifications'

export default class App extends Component {
  render () {
    return (
      <Router>
        <div id="app">
          <Helmet title="Howdju" />

          <Header/>

          <Route exact path="/" component={Home} />
          <Route path="/s/:statementId/:statementSlug" component={StatementJustifications} />

        </div>
      </Router>
    )
  }
}
