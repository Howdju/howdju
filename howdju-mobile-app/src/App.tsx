import React, { useCallback, useEffect, useState } from "react";
import { useColorScheme } from "react-native";
import { NavigationContainer, DefaultTheme } from "@react-navigation/native";
import { createBottomTabNavigator } from "@react-navigation/bottom-tabs";
import ShareMenu, { ShareResponse } from "react-native-share-menu";
import MaterialCommunityIcons from "react-native-vector-icons/MaterialCommunityIcons";
import {
  adaptNavigationTheme,
  Provider as PaperProvider,
} from "react-native-paper";

import BrowserScreen from "@/screens/BrowserScreen";
import ShareDebugScreen from "@/screens/ShareDebugScreen";
import SettingsScreen from "./screens/SettingsScreen";
import { darkTheme, lightTheme } from "./themes";
import AppSettings from "./AppSettings";
import { SafeAreaProvider } from "react-native-safe-area-context";

const Tab = createBottomTabNavigator();

const EMPTY_SHARE_RESPONSE: ShareResponse = { items: [] };

const App = (): JSX.Element => {
  const [shareResponse, setShareResponse] = useState(EMPTY_SHARE_RESPONSE);

  const handleShare = useCallback((response?: ShareResponse) => {
    if (!response) {
      return;
    }

    setShareResponse(response);
  }, []);

  useEffect(() => {
    ShareMenu.getInitialShare(handleShare);
  }, [handleShare]);

  useEffect(() => {
    const listener = ShareMenu.addNewShareListener(handleShare);
    return () => {
      listener.remove();
    };
  }, [handleShare]);

  const items = shareResponse?.items;
  const extraData = shareResponse?.extraData as Record<string, unknown>;
  const isDark = useColorScheme() === "dark";
  const theme = isDark ? darkTheme : lightTheme;

  const { LightTheme: lightNavigationTheme, DarkTheme: darkNavigationTheme } =
    adaptNavigationTheme({
      reactNavigationLight: DefaultTheme,
      reactNavigationDark: DefaultTheme,
      materialLight: lightTheme,
      materialDark: darkTheme,
    });

  return (
    <PaperProvider theme={theme}>
      <AppSettings>
        <SafeAreaProvider>
          <NavigationContainer
            theme={isDark ? darkNavigationTheme : lightNavigationTheme}
          >
            <Tab.Navigator>
              {/* TODO(62): ensure using a render callback does not introduce
            performance issues
            https://reactnavigation.org/docs/hello-react-navigation/#passing-additional-props
        */}
              <Tab.Screen
                name="Browser"
                options={{
                  tabBarLabel: "Browser",
                  headerShown: false,
                  tabBarIcon: function TabBarIcon({ color, size, focused }) {
                    return (
                      <MaterialCommunityIcons
                        name="web"
                        color={color}
                        size={size}
                        style={{ fontWeight: focused ? "bold" : "normal" }}
                      />
                    );
                  },
                }}
              >
                {(props) => <BrowserScreen {...props} items={items} />}
              </Tab.Screen>
              <Tab.Screen
                name="ShareDebug"
                options={{
                  tabBarLabel: "Debug",
                  tabBarIcon: function TabBarIcon({ color, size, focused }) {
                    return (
                      <MaterialCommunityIcons
                        name={focused ? "bug" : "bug-outline"}
                        color={color}
                        size={size}
                      />
                    );
                  },
                }}
              >
                {(props) => (
                  <ShareDebugScreen
                    {...props}
                    items={items}
                    extraData={extraData}
                  />
                )}
              </Tab.Screen>
              <Tab.Screen
                name="Settings"
                component={SettingsScreen}
                options={{
                  tabBarLabel: "Settings",
                  tabBarIcon: function TabBarIcon({ color, size, focused }) {
                    return (
                      <MaterialCommunityIcons
                        name={focused ? "cog" : "cog-outline"}
                        color={color}
                        size={size}
                      />
                    );
                  },
                }}
              />
            </Tab.Navigator>
          </NavigationContainer>
        </SafeAreaProvider>
      </AppSettings>
    </PaperProvider>
  );
};

export default App;
