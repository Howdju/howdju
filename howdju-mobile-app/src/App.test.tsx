import "react-native";
import React from "react";
import App from "../src/App";

// Warning: TestRenderer must be required after react-native.
import { render, waitFor } from "@testing-library/react-native";

jest.mock("react-native", () => {
  const RN = jest.requireActual("react-native"); // use original implementation, which comes with mocks out of the box

  // mock modules/components created by assigning to NativeModules
  RN.NativeModules.ShareMenu = {
    getSharedText: jest.fn(),
    // addNewShareListener: jest.fn(),
    addListener: jest.fn(),
    removeListeners: jest.fn(),
  };

  // mock modules created through UIManager
  // RN.UIManager.getViewManagerConfig = (name: string) => {
  //   if (name === 'SomeNativeModule') {
  //     return {someMethod: jest.fn()};
  //   }
  //   return {};
  // };
  return RN;
});

describe("App", () => {
  it("renders correctly", async () => {
    const result = render(<App />);
    await waitFor(() => result.getAllByText("Browser"));
    expect(result.toJSON()).toMatchSnapshot();
  });
});
