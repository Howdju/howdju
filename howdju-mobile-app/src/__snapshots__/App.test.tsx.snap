// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        [
          {
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenNavigationContainer
        style={
          {
            "flex": 1,
            "overflow": "hidden",
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          gestureResponseDistance={
            {
              "bottom": -1,
              "end": -1,
              "start": -1,
              "top": -1,
            }
          }
          nativeID="animatedComponent"
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 0,
            }
          }
        >
          <View
            accessibilityElementsHidden={false}
            importantForAccessibility="auto"
            style={
              [
                {
                  "backgroundColor": "rgb(252, 252, 255)",
                  "flex": 1,
                },
                [
                  {
                    "flex": 1,
                    "flexDirection": "column-reverse",
                  },
                  undefined,
                ],
              ]
            }
          >
            <View
              style={
                {
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  [
                    {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    undefined,
                  ]
                }
              >
                <RNCWebView
                  cacheEnabled={true}
                  injectedJavaScriptBeforeContentLoadedForMainFrameOnly={true}
                  injectedJavaScriptForMainFrameOnly={true}
                  javaScriptEnabled={true}
                  messagingEnabled={false}
                  onContentProcessDidTerminate={[Function]}
                  onError={[Function]}
                  onHttpError={[Function]}
                  onLoadingError={[Function]}
                  onLoadingFinish={[Function]}
                  onLoadingProgress={[Function]}
                  onLoadingStart={[Function]}
                  onMessage={[Function]}
                  onRenderProcessGone={[Function]}
                  onShouldStartLoadWithRequest={[Function]}
                  source={
                    {
                      "uri": "https://www.howdju.com/recent-activity/",
                    }
                  }
                  style={
                    [
                      {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      {
                        "backgroundColor": "#ffffff",
                      },
                      undefined,
                    ]
                  }
                  useSharedProcessPool={true}
                />
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                collapsable={false}
                nativeID="animatedComponent"
                pointerEvents="box-none"
                style={
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "zIndex": 0,
                  }
                }
              >
                <View
                  collapsable={false}
                  nativeID="animatedComponent"
                  style={
                    {
                      "backgroundColor": "rgb(232, 240, 247)",
                      "borderBottomColor": "rgb(114, 119, 127)",
                      "flex": 1,
                      "shadowColor": "rgb(114, 119, 127)",
                      "shadowOffset": {
                        "height": 0.5,
                        "width": 0,
                      },
                      "shadowOpacity": 0.85,
                      "shadowRadius": 0,
                    }
                  }
                />
              </View>
              <View
                collapsable={false}
                nativeID="animatedComponent"
                pointerEvents="box-none"
                style={
                  {
                    "height": 44,
                    "maxHeight": undefined,
                    "minHeight": undefined,
                    "opacity": undefined,
                    "transform": undefined,
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    {
                      "height": 0,
                    }
                  }
                />
                <View
                  pointerEvents="box-none"
                  style={
                    {
                      "alignItems": "stretch",
                      "flex": 1,
                      "flexDirection": "row",
                    }
                  }
                >
                  <View
                    collapsable={false}
                    nativeID="animatedComponent"
                    pointerEvents="box-none"
                    style={
                      {
                        "alignItems": "flex-start",
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "justifyContent": "center",
                        "marginStart": 0,
                      }
                    }
                  />
                  <View
                    collapsable={false}
                    nativeID="animatedComponent"
                    pointerEvents="box-none"
                    style={
                      {
                        "justifyContent": "center",
                        "marginHorizontal": 16,
                        "maxWidth": 718,
                      }
                    }
                  >
                    <Text
                      accessibilityRole="header"
                      aria-level="1"
                      collapsable={false}
                      nativeID="animatedComponent"
                      numberOfLines={1}
                      style={
                        {
                          "color": "rgb(26, 28, 30)",
                          "fontSize": 17,
                          "fontWeight": "600",
                        }
                      }
                    >
                      Browser
                    </Text>
                  </View>
                  <View
                    collapsable={false}
                    nativeID="animatedComponent"
                    pointerEvents="box-none"
                    style={
                      {
                        "alignItems": "flex-end",
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "justifyContent": "center",
                        "marginEnd": 0,
                      }
                    }
                  />
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenNavigationContainer>
      <View
        collapsable={false}
        nativeID="animatedComponent"
        onLayout={[Function]}
        pointerEvents="auto"
        style={
          {
            "backgroundColor": "rgb(232, 240, 247)",
            "borderTopColor": "rgb(114, 119, 127)",
            "borderTopWidth": 0.5,
            "bottom": 0,
            "elevation": 8,
            "height": 49,
            "left": 0,
            "paddingBottom": 0,
            "paddingHorizontal": 0,
            "position": null,
            "right": 0,
            "transform": [
              {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
        <View
          accessibilityRole="tablist"
          style={
            {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            accessibilityLabel="Browser, tab, 1 of 3"
            accessibilityRole="button"
            accessibilityState={
              {
                "selected": true,
              }
            }
            accessibilityStates={
              [
                "selected",
              ]
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "alignItems": "center",
                  "flex": 1,
                },
                {
                  "backgroundColor": "transparent",
                },
                {
                  "flexDirection": "column",
                  "justifyContent": "flex-end",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                [
                  {
                    "flex": 1,
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "height": "100%",
                      "justifyContent": "center",
                      "minWidth": 25,
                      "position": "absolute",
                      "width": "100%",
                    },
                    {
                      "opacity": 1,
                    },
                  ]
                }
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    [
                      {
                        "color": "rgb(0, 99, 156)",
                        "fontSize": 25,
                      },
                      {
                        "fontWeight": "bold",
                      },
                      {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      {},
                    ]
                  }
                >
                  󰖟
                </Text>
              </View>
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "height": "100%",
                      "justifyContent": "center",
                      "minWidth": 25,
                      "position": "absolute",
                      "width": "100%",
                    },
                    {
                      "opacity": 0,
                    },
                  ]
                }
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    [
                      {
                        "color": "#81868B",
                        "fontSize": 25,
                      },
                      {
                        "fontWeight": "normal",
                      },
                      {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      {},
                    ]
                  }
                >
                  󰖟
                </Text>
              </View>
            </View>
            <Text
              numberOfLines={1}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "textAlign": "center",
                  },
                  {
                    "color": "rgb(0, 99, 156)",
                  },
                  {
                    "fontSize": 10,
                  },
                  undefined,
                ]
              }
            >
              Browser
            </Text>
          </View>
          <View
            accessibilityLabel="Debug, tab, 2 of 3"
            accessibilityRole="button"
            accessibilityState={
              {
                "selected": false,
              }
            }
            accessibilityStates={[]}
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "alignItems": "center",
                  "flex": 1,
                },
                {
                  "backgroundColor": "transparent",
                },
                {
                  "flexDirection": "column",
                  "justifyContent": "flex-end",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                [
                  {
                    "flex": 1,
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "height": "100%",
                      "justifyContent": "center",
                      "minWidth": 25,
                      "position": "absolute",
                      "width": "100%",
                    },
                    {
                      "opacity": 0,
                    },
                  ]
                }
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    [
                      {
                        "color": "rgb(0, 99, 156)",
                        "fontSize": 25,
                      },
                      undefined,
                      {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      {},
                    ]
                  }
                >
                  󰃤
                </Text>
              </View>
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "height": "100%",
                      "justifyContent": "center",
                      "minWidth": 25,
                      "position": "absolute",
                      "width": "100%",
                    },
                    {
                      "opacity": 1,
                    },
                  ]
                }
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    [
                      {
                        "color": "#81868B",
                        "fontSize": 25,
                      },
                      undefined,
                      {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      {},
                    ]
                  }
                >
                  󰨰
                </Text>
              </View>
            </View>
            <Text
              numberOfLines={1}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "textAlign": "center",
                  },
                  {
                    "color": "#81868B",
                  },
                  {
                    "fontSize": 10,
                  },
                  undefined,
                ]
              }
            >
              Debug
            </Text>
          </View>
          <View
            accessibilityLabel="Settings, tab, 3 of 3"
            accessibilityRole="button"
            accessibilityState={
              {
                "selected": false,
              }
            }
            accessibilityStates={[]}
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "alignItems": "center",
                  "flex": 1,
                },
                {
                  "backgroundColor": "transparent",
                },
                {
                  "flexDirection": "column",
                  "justifyContent": "flex-end",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                [
                  {
                    "flex": 1,
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "height": "100%",
                      "justifyContent": "center",
                      "minWidth": 25,
                      "position": "absolute",
                      "width": "100%",
                    },
                    {
                      "opacity": 0,
                    },
                  ]
                }
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    [
                      {
                        "color": "rgb(0, 99, 156)",
                        "fontSize": 25,
                      },
                      undefined,
                      {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      {},
                    ]
                  }
                >
                  󰒓
                </Text>
              </View>
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "height": "100%",
                      "justifyContent": "center",
                      "minWidth": 25,
                      "position": "absolute",
                      "width": "100%",
                    },
                    {
                      "opacity": 1,
                    },
                  ]
                }
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    [
                      {
                        "color": "#81868B",
                        "fontSize": 25,
                      },
                      undefined,
                      {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      {},
                    ]
                  }
                >
                  󰢻
                </Text>
              </View>
            </View>
            <Text
              numberOfLines={1}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "textAlign": "center",
                  },
                  {
                    "color": "#81868B",
                  },
                  {
                    "fontSize": 10,
                  },
                  undefined,
                ]
              }
            >
              Settings
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
